### PR TITLE
configured spring; added user model with dao and service

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -7,6 +7,50 @@
...
+            <version>5.4.21.Final</version>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.21</version>
+        </dependency>
+    </dependencies>
 
 </project>
```
Comment (tarkaiev) [inline pom.xml]: add empty line pls

**Review by tarkaiev — COMMENTED** [review]
good job

File: pom.xml [hunk 2]
```diff
@@ -7,6 +7,50 @@
     <groupId>org.example</groupId>
     <artifactId>spring_project</artifactId>
     <version>1.0-SNAPSHOT</version>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>7</source>
```
Comment (Sofasmile) [inline pom.xml]: better use 8 or 11 version

File: src/main/java/spring/intro/Main.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package spring.intro;
+
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import spring.intro.config.AppConfig;
+import spring.intro.model.User;
+import spring.intro.service.UserService;
+
+public class Main {
+    public static void main(String[] args) {
+        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
```
Comment (Sofasmile) [inline src/main/java/spring/intro/Main.java]: What is the length of this line?

File: src/main/java/spring/intro/Main.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        UserService userService = context.getBean(UserService.class);
+        userService.add(new User("user1", "111"));
+        userService.add(new User("user2", "222"));
+        userService.add(new User("user3", "333"));
+        System.out.println(userService.listUsers());
```
Comment (Sofasmile) [inline src/main/java/spring/intro/Main.java:16]: does it work correctly?

File: src/main/java/spring/intro/dao/impl/UserDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.stereotype.Repository;
+import spring.intro.dao.UserDao;
+import spring.intro.exceptions.DataProcessingException;
+import spring.intro.model.User;
+import java.util.List;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    @Autowired
+    private SessionFactory sessionFactory;
```
Comment (Sofasmile) [inline src/main/java/spring/intro/dao/impl/UserDaoImpl.java]: What is better: setter auto wiring, constructor auto wiring, or field injection?

File: src/main/java/spring/intro/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.stereotype.Service;
+import spring.intro.dao.UserDao;
+import spring.intro.model.User;
+import spring.intro.service.UserService;
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Autowired
+    private UserDao userDao;
```
Comment (Sofasmile) [inline src/main/java/spring/intro/service/impl/UserServiceImpl.java]: What is better: setter, constructor, or field injection?

File: pom.xml (hunk 1 above)
Comment (TimoxaV) [reply pom.xml]: done

File: pom.xml (hunk 2 above)
Comment (TimoxaV) [reply pom.xml]: done

File: src/main/java/spring/intro/Main.java (hunk 3 above)
Comment (TimoxaV) [reply src/main/java/spring/intro/Main.java]: 101 chars. But clean package says that build was successful...
I split it in two lines

File: src/main/java/spring/intro/Main.java (hunk 4 above)
Comment (TimoxaV) [reply src/main/java/spring/intro/Main.java:16]: well, yes, I can see the list and all its components in appropriate way.

File: src/main/java/spring/intro/dao/impl/UserDaoImpl.java (hunk 5 above)
Comment (TimoxaV) [reply src/main/java/spring/intro/dao/impl/UserDaoImpl.java]: constructor. changed this part.

File: src/main/java/spring/intro/service/impl/UserServiceImpl.java (hunk 6 above)
Comment (TimoxaV) [reply src/main/java/spring/intro/service/impl/UserServiceImpl.java]: constructor. changed.

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**